src/python/pants/util/dirutil.py

```diff
@@ -19,6 +19,23 @@
 from pants.util.strutil import ensure_text
 
 
+def longest_dir_prefix(path, prefixes):
```

Comment:
This is the kind of thing that would be easiest accomplished with the Trie that is used for source_roots.